### PR TITLE
[FX-317] Part 2: Add log statements

### PR DESCRIPTION
**Pull request by djoksimo**
## What

- Add log statements with helpful attributes (context), prefixes
- Logs all HTTP requests and responses
- Logs all errors with level "error"

## Why

Learn more: https://docs.google.com/document/d/1NYe6JtnBJAVrtjIKREdwoBt3lqh18vfXqzj9EaodXhI/edit#bookmark=id.2rdd25jqml4p

## Test Plan

DD logs https://app.datadoghq.com/logs?query=service%3Aios-sdk%20&cols=host%2Cservice&index=%2A&messageDisplay=inline&stream_sort=desc&viz=stream&from_ts=1690904666204&to_ts=1690904713184&live=false

image

- ✅  iOS QA Tests passed locally
- ✅  Unit Tests passed locally

## How

This PR Stack (FX-317) can be considered a patch-level change.
This PR is based off of #54

### CODE REVIEW

**Comment by djoksimo** [comment]
Current dependencies on/for this PR:
* main
  * **PR #54** <a href="https://app.graphite.dev/github/pr/teamforage/forage-ios-sdk/54" target="_blank">Graphite</a> 
    * **PR #55** <a href="https://app.graphite.dev/github/pr/teamforage/forage-ios-sdk/55" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/teamforage/forage-ios-sdk/55?utm_source=stack-comment).

**Review by dleis612 — APPROVED** [review]

**Review by devinmorgan — CHANGES_REQUESTED** [review]
Thanks for adding all of these logs! I wanted to clarify whether this PR needs to be updated to take advantage of your most recent commits in #54? I noticed
1. references to `sharedLogger`
2. passing `error: nil` and `attributes: nil` (I thought these params were optional now)

**Comment by djoksimo** [comment]
> 2\. passing `error: nil` and `attributes: nil` (I thought these params were optional now)

@devinmorgan it seems like this is easier said than done in swift when I set logger type to `ForageLogger` instead of `DatadogLogger`.

I will ask the Symphony team about this.

**Comment by devinmorgan** [comment]
Sounds good! Happy to approve this though. It can be a follow up diff

**Review by dleis612 — APPROVED** [review]

File: Sources/ForageSDK/Foundation/Telemetry/ForageLogger.swift [hunk 1]
```diff
@@ -13,35 +13,75 @@ internal struct ForageLogContext {
...
 
-protocol ForageLogger {
-    init(_ config: ForageLoggerConfig)
+internal protocol ForageLogger {
+    init(_ config: ForageLoggerConfig?)
     
+    /// Adds additional context to the logger, providing useful information for each log message.
+    /// - Parameter newContext: The ForageLogContext to be added to the logger.
+    /// - Returns: The instance of the ForageLogger with the additional context applied. Useful for method chaining.
+    func addContext(_ newContext: ForageLogContext) -> ForageLogger
```
Comment (devinmorgan) [inline Sources/ForageSDK/Foundation/Telemetry/ForageLogger.swift:37]: I love this pattern! Immutable and clean - nice work!!

File: Sources/ForageSDK/ForageSDK.swift [hunk 2]
```diff
@@ -38,10 +39,21 @@ public class ForageSDK {
...
+                prefix: ""
+            )
+        )
+        self.logger = logger
+        LDManager.shared.initialize(self.environment, logger: logger)
         VGSCollectLogger.shared.disableAllLoggers()
-        self.service = LiveForageService()
+        
+        let provider = Provider(logger: logger)
+        self.service = LiveForageService(provider: provider, logger: logger)
```
Comment (devinmorgan) [inline Sources/ForageSDK/ForageSDK.swift:56]: love this! Instantiate a logger at the top level and pass as a dependency down to keep services pure. Great work!

**Review by devinmorgan — APPROVED** [review]
These latest changes include some great design decisions! Great work Danilo

**Pull request merged**